All tests are now passing except test: `ci/prow/manifest-diff` and  is failing with error:
```
hack/manifest-diff-upstream.sh failed. Permissions not equivalent in config/vpa/vpa-rbac.yaml and /tmp/manifest-diff.kaylThaPUM/upstream/vpa-rbac.yaml
If changes are made to /tmp/manifest-diff.kaylThaPUM/upstream/vpa-rbac.yaml, equivalent changes should be made to config/vpa/vpa-rbac.yaml
If OpenShift-specific changes are made to config/vpa/vpa-rbac.yaml, those changes should be represented in hack/filter-upstream-rbac.jq so that a transformed upstream rbac will match the downstream changed version.
make: *** [Makefile:134: manifest-diff] Error 1
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:84","func":"sigs.k8s.io/prow/pkg/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2026-06-19T18:36:46Z"} 
```
_This seems reasonable to me, as it is validating `config/vpa/vpa-rbac.yaml` against the upstream RBAC definitions. 
The objective of this issue is to address the additional privileged RBAC permissions that were previously synchronized from upstream, so the current behavior aligns with that goal. 
With this fix in place, we also have an opportunity to revisit and enhance our test coverage._

**Update**: My bad, we should still design the `hack/manifest-diff-upstream.sh` script in such a way that all changes to the upstream RBACs should be flagged/synchronised with the current `vpa-rbac.yaml` file. In the next commit I will make the changes accordingly.

--- a/bundle/manifests/vertical-pod-autoscaler.clusterserviceversion.yaml
+++ b/bundle/manifests/vertical-pod-autoscaler.clusterserviceversion.yaml
@@ -56,7 +56,7 @@ metadata:
     categories: OpenShift Optional
     certifiedLevel: Primed
     containerImage: quay.io/openshift/origin-vertical-pod-autoscaler-operator:4.22.0
-    createdAt: "2026-06-20T06:10:09Z"
+    createdAt: "2026-06-24T14:04:09Z"
     description: An operator to run the OpenShift Vertical Pod Autoscaler. Vertical
       Pod Autoscaler (VPA) can be configured to monitor a workload's resource utilization,
       and then adjust its CPU and memory limits by updating the pod (future) or restarting
@@ -395,6 +395,13 @@ spec:
           - list
           - watch
         - apiGroups:
+          - '*'
+          resources:
+          - '*/scale'
+          verbs:
+          - get
+          - watch
+        - apiGroups:
           - ""
           resources:
           - replicationcontrollers
@@ -460,6 +467,8 @@ spec:
           - get
           - list
           - watch
+          - patch
+          - update
         - apiGroups:
           - poc.autoscaling.k8s.io
           resources:
@@ -512,6 +521,13 @@ spec:
           verbs:
           - get
           - patch
+        - apiGroups:
+          - '*'
+          resources:
+          - '*/scale'
+          verbs:
+          - get
+          - watch
         - apiGroups:
           - ""
           resources:
@@ -571,6 +587,8 @@ spec:
           - get
           - list
           - watch
+          - patch
+          - update
         - apiGroups:
           - poc.autoscaling.k8s.io
           resources:
@@ -607,6 +625,13 @@ spec:
           verbs:
           - get
           - list
+          - watch
+        - apiGroups:
+          - '*'
+          resources:
+          - '*/scale'
+          verbs:
+          - get
           - watch
         - apiGroups:
           - ""

--- a/bundle/manifests/vertical-pod-autoscaler.clusterserviceversion.yaml
+++ b/bundle/manifests/vertical-pod-autoscaler.clusterserviceversion.yaml
@@ -56,7 +56,7 @@ metadata:
     categories: OpenShift Optional
     certifiedLevel: Primed
     containerImage: quay.io/openshift/origin-vertical-pod-autoscaler-operator:4.22.0
-    createdAt: "2026-06-19T14:27:08Z"
+    createdAt: "2026-06-20T06:10:09Z"
     description: An operator to run the OpenShift Vertical Pod Autoscaler. Vertical
       Pod Autoscaler (VPA) can be configured to monitor a workload's resource utilization,
       and then adjust its CPU and memory limits by updating the pod (future) or restarting
@@ -648,8 +648,8 @@ spec:
         - apiGroups:
           - ""
           resources:
-          - pods
           - pods/resize
+          - pods
           verbs:
           - patch
         serviceAccountName: vpa-updater

--- a/config/vpa/vpa-rbac.yaml
+++ b/config/vpa/vpa-rbac.yaml
@@ -129,8 +129,8 @@ rules:
 - apiGroups:
   - ""
   resources:
-  - pods
   - pods/resize
+  - pods
   verbs:
   - patch
 ---

--- a/config/vpa/vpa-rbac.yaml
+++ b/config/vpa/vpa-rbac.yaml
@@ -37,6 +37,8 @@ rules:
       - get
       - list
       - watch
+      - patch
+      - update
   - apiGroups:
       - "poc.autoscaling.k8s.io"
     resources:
@@ -207,6 +209,13 @@ kind: ClusterRole
 metadata:
   name: system:vpa-target-reader
 rules:
+  - apiGroups:
+    - '*'
+    resources:
+    - '*/scale'
+    verbs:
+    - get
+    - watch
   - apiGroups:
       - ""
     resources:

--- a/hack/filter-upstream-rbac.jq
+++ b/hack/filter-upstream-rbac.jq
@@ -3,12 +3,6 @@
         apiGroups: [ "apps.openshift.io" ],
         resources: [ "deploymentconfigs", "deploymentconfigs/scale" ],
         verbs: [ "get", "list", "watch" ] } ] |
-# Security fix: Remove wildcard apiGroups from vpa-target-reader
-(.items[] | select(.kind=="ClusterRole" and .metadata.name=="system:vpa-target-reader")).rules |=
-  map(select(.apiGroups != ["*"])) |
-# Security fix: Remove patch/update from events in vpa-actor
-(.items[] | select(.kind=="ClusterRole" and .metadata.name=="system:vpa-actor")).rules |=
-  map(if .resources == ["events"] then .verbs |= (. - ["patch", "update"]) else . end) |
 # Security fix: Split webhook config permissions with resourceNames
 (.items[] | select(.kind=="ClusterRole" and .metadata.name=="system:vpa-admission-controller")).rules |= (
   map(

--- a/hack/filter-upstream-rbac.jq
+++ b/hack/filter-upstream-rbac.jq
@@ -10,7 +10,7 @@
 (.items[] | select(.kind=="ClusterRole" and .metadata.name=="system:vpa-actor")).rules |=
   map(if .resources == ["events"] then .verbs |= (. - ["patch", "update"]) else . end) |
 # Security fix: Split webhook config permissions with resourceNames
-(.items[] | select(.kind=="ClusterRole" and .metadata.name=="system:vpa-admission-controller")).rules |=
+(.items[] | select(.kind=="ClusterRole" and .metadata.name=="system:vpa-admission-controller")).rules |= (
   map(
     if (.apiGroups == ["admissionregistration.k8s.io"] and .resources == ["mutatingwebhookconfigurations"]) then
       [
@@ -18,6 +18,7 @@
         {apiGroups: .apiGroups, resources: .resources, resourceNames: ["vpa-webhook-config"], verbs: ["delete", "patch", "update"]}
       ]
     else . end
-  ) | flatten |
+  ) | flatten
+) |
 # We use namespace openshift-vertical-pod-autoscaler instead of kube-system. Replace all namespaces
 walk(if type == "object" and has("namespace") then .namespace="openshift-vertical-pod-autoscaler" else . end)

--- a/hack/yamls2list.sed
+++ b/hack/yamls2list.sed
@@ -1,7 +1,11 @@
 # Transform concatenated yaml files of k8s objects into a k8s List
 # Add kind, metadata and items at the top
-1 i \
----\nkind: List\nmetadata: {}\napiVersion: v1\nitems:
+1 i\
+---\
+kind: List\
+metadata: {}\
+apiVersion: v1\
+items:
 # indent every existing line by 2 spaces
 s/^/  /
 # Any line which was originally "---" should be deleted and the following line should begin with "- " since it's the first line of an item in the list

